src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -249,24 +282,81 @@ private CSharpAttributeData GetAttribute(BoundAttribute boundAttribute, BindingD
 
             Debug.Assert(!constructorArgsArray.IsDefault, "Property of VisitArguments");
 
-            ImmutableArray<int> constructorArgumentsSourceIndices;
-            ImmutableArray<TypedConstant> constructorArguments;
+            ImmutableArray<int> argsToParamsOpt = boundAttribute.ConstructorArgumentsToParamsOpt;
+            ImmutableArray<TypedConstant> rewrittenArguments;
             if (hasErrors || attributeConstructor.ParameterCount == 0)
             {
-                constructorArgumentsSourceIndices = default(ImmutableArray<int>);
-                constructorArguments = constructorArgsArray;
+                rewrittenArguments = constructorArgsArray;
```

Comment:
>rewrittenArguments = constructorArgsArray [](http://example.com/codeflow?start=16&length=41)

Is it possible to get here with a `params` array attribute constructor where the length of `rewrittenArguments` does not match the constructor parameter length? Do consumers of the `SourceAttributeData` expect the lengths to match?

Reply:
I'll write a test against the compiler prior to this PR and ensure we haven't changed the behavior.

Reply:
Added `AttributeTests.TestBadParamsCtor`